tests/Unit/Evolution/Systems/Cce/Actions/Test_Psi0Matching.cpp

```diff
@@ -0,0 +1,253 @@
+  static std::tuple<db::DataBox<DbTags>&&> apply(
+      db::DataBox<DbTags>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    db::mutate_apply<GaugeUpdateAngularFromCartesian<
+        Tags::PartiallyFlatAngularCoords, Tags::PartiallyFlatCartesianCoords>>(
+        make_not_null(&box));
+    db::mutate_apply<GaugeUpdateInterpolator<Tags::PartiallyFlatAngularCoords>>(
+        make_not_null(&box));
+    return {std::move(box)};
+  }
+};
+
+using real_tags_to_generate = tmpl::list<Tags::PartiallyFlatCartesianCoords>;
+
+using real_tags_to_compute = tmpl::list<Tags::PartiallyFlatAngularCoords>;
+
+using swsh_volume_tags_to_generate = tmpl::list<Tags::BondiJ, Tags::OneMinusY>;
+
+using swsh_boundary_tags_to_generate =
+    tmpl::list<Tags::CauchyGaugeC, Tags::CauchyGaugeD, Tags::CauchyGaugeOmega,
+               Tags::BoundaryValue<Tags::BondiR>,
+               Tags::BoundaryValue<Tags::BondiBeta>>;
+
+using swsh_volume_tags_to_compute =
+    tmpl::list<Tags::BondiJCauchyView, Tags::Dy<Tags::BondiJCauchyView>,
+               Tags::Dy<Tags::Dy<Tags::BondiJCauchyView>>, Tags::Psi0Match,
+               Tags::Dy<Tags::Psi0Match>>;
+
+using swsh_boundary_tags_to_compute =
+    tmpl::list<Tags::BoundaryValue<Tags::Psi0Match>,
+               Tags::BoundaryValue<Tags::Dlambda<Tags::Psi0Match>>>;
+
+template <typename Metavariables>
+struct mock_characteristic_evolution {
+  using simple_tags = tmpl::push_back<
+      db::AddSimpleTags<
+          ::Tags::Variables<
+              tmpl::append<real_tags_to_generate, real_tags_to_compute>>,
+          ::Tags::Variables<tmpl::append<swsh_volume_tags_to_generate,
+                                         swsh_volume_tags_to_compute>>,
+          ::Tags::Variables<tmpl::append<swsh_boundary_tags_to_generate,
+                                         swsh_boundary_tags_to_compute>>>,
+      Spectral::Swsh::Tags::SwshInterpolator<Tags::PartiallyFlatAngularCoords>>;
+
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<simple_tags>>>,
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Evolve,
+          tmpl::list<UpdateCartesianFromAngular, Actions::CalculatePsi0>>>;
+};
+
+struct metavariables {
+  using component_list =
+      tmpl::list<mock_characteristic_evolution<metavariables>>;
+  enum class Phase { Initialization, Evolve, Exit };
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.Cce.Actions.Psi0Matching",
+                  "[Unit][Cce]") {
```

Comment:
It would be nice to have a comment somewhere explaining what this test actually does.
Here's what it looks like it does to me (is this correct?):
  * Fills a bunch of variables with random numbers (filtered so that there is no aliasing in highest modes)
  * Puts those variables in two places: the MockRuntimeSystem and a databox called expected_box.
  * Calls next_action twice (which fills the results in the databox of MockRuntimeSystem)
  * Calls all the individual mutators by hand, filling the results in expected_box
  * Compares MockRuntimeSystem's databox vs expected_box.
So if any of the individual mutators had an error, this test wouldn't catch it (because the erroneous mutator would be called twice and then compared), but what is tested is the automatic calling chain of next_action.  This is probably fine (because Test_NewmanPenrose.cpp tests for correctness), but it would be nice to see this explained.

Reply:
It's already a good summary. So I just copy and paste it to the beginning of the test, and put each step in front of the corresponding code block.